File: services/auth-service/src/routes/health.py
"""
Health Check Routes
"""

from flask import Blueprint, jsonify
from datetime import datetime
import time

health_bp = Blueprint('health', __name__)

START_TIME = time.time()

@health_bp.route('/health')
def health():
    """Liveness probe"""
    return jsonify({
        'status': 'healthy',
        'service': 'auth-service',
        'timestamp': datetime.utcnow().isoformat(),
        'uptime': int(time.time() - START_TIME)
    })

@health_bp.route('/ready')
def ready():
    """Readiness probe - check database connection"""
    from src.database import db
    
    try:
        db.session.execute(db.text('SELECT 1'))
        db_status = 'up'
    except Exception as e:
        db_status = f'down: {str(e)}'
    
    is_ready = db_status == 'up'
    
    return jsonify({
        'status': 'ready' if is_ready else 'not ready',
        'service': 'auth-service',
        'timestamp': datetime.utcnow().isoformat(),
        'dependencies': {
            'database': db_status
        }
    }), 200 if is_ready else 503

File: services/order-service/src/routes/health.js
/**
 * Health Check Routes
 */

const express = require('express');
const router = express.Router();
const dynamodb = require('../utils/dynamodb');

const startTime = Date.now();

router.get('/health', (req, res) => {
    res.json({
        status: 'healthy',
        service: 'order-service',
        timestamp: new Date().toISOString(),
        uptime: Math.floor((Date.now() - startTime) / 1000)
    });
});

router.get('/ready', async (req, res) => {
    let dbStatus = 'unknown';

    try {
        await dynamodb.checkConnection();
        dbStatus = 'up';
    } catch (error) {
        dbStatus = `down: ${error.message}`;
    }

    const isReady = dbStatus === 'up';

    res.status(isReady ? 200 : 503).json({
        status: isReady ? 'ready' : 'not ready',
        service: 'order-service',
        timestamp: new Date().toISOString(),
        dependencies: {
            dynamodb: dbStatus
        }
    });
});

module.exports = router;
